src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1517,8 +1517,16 @@ void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
 
         assert((simdSize == 8) || (simdSize == 12) || (simdSize == 16) || (simdSize == 32));
 
-        if ((argCnt == 1) ||
-            ((vecCns.i64[0] == vecCns.i64[1]) && ((simdSize <= 16) || (vecCns.i64[2] == vecCns.i64[3]))))
+        bool allBytesAreSame = true;
+        for (UINT32 i = 0; i < simdSize; i++)
```

Comment:
Can this loop over `i32`/`i64` elements to keep the number of comparisons to a minimum?

Reply:
Sure!